patches/net/minecraft/world/entity/npc/Villager.java.patch

```diff
@@ -37,12 +37,15 @@
                  p_35409_.addFreshEntityWithPassengers(witch);
                  this.releaseAllPois();
                  this.discard();
-@@ -831,7 +_,7 @@
+@@ -831,7 +_,10 @@
      @Override
      public boolean wantsToPickUp(ItemStack p_35543_) {
          Item item = p_35543_.getItem();
 -        return (WANTED_ITEMS.contains(item) || this.getVillagerData().getProfession().requestedItems().contains(item))
-+        return (WANTED_ITEMS.contains(item) || this.getVillagerData().getProfession().requestedItems().contains(item) || (p_35543_.getItem() instanceof net.neoforged.neoforge.common.SpecialPlantable specialPlantable && specialPlantable.villagerCanPlantItem(this)))
++        // Neo: Patched so that Farmer Villagers will pick up SpecialPlantable items. Also fixes MC-274244 by making "minecraft:villager_plantable_seeds" tagged items also be picked up by Farmer Villagers to be planted later.
++        boolean isFarmerDesiredSeed = (p_35543_.is(ItemTags.VILLAGER_PLANTABLE_SEEDS) || (p_35543_.getItem() instanceof net.neoforged.neoforge.common.SpecialPlantable specialPlantable && specialPlantable.villagerCanPlantItem(this)))
++                && this.getVillagerData().getProfession().secondaryPoi().stream().anyMatch(secondaryPoi -> secondaryPoi.defaultBlockState().is(net.neoforged.neoforge.common.Tags.Blocks.VILLAGER_FARMLANDS));
```

Comment:
This seems like a *potential* break in behavior, where previously before this patch villagers would pickup anything that `villagerCanPlantItem` is true for, but now would only take effect if the villager's secondary POI is the farmland block. 

Should we preserve the previous behavior (in effect removing the secondary POI check), or was it a bug to be fixed?

Reply:
The `villagerCanPlantItem` was supposed to be for mainly-farmland plantable seeds. Farmers (or other villagers who track farmland) are pretty much the only ones that can plant items and is what people would expect from `villagerCanPlantItem`. For example, if I have a special crop seed and return true for `villagerCanPlantItem` and it is picked up by a modded villager who doesn't place plants or doesn't track farmland, that item goes into the villager's inventory and stays stuck there forever. Not good.

Reply:
So, a bug then. It should have been noted in the PR description, alongside the fix to include `minecraft:villager_plantable_seeds`, but at least it's now documented in this thread.